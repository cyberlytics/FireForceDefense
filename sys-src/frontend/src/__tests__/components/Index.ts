import { mount, RouterLinkStub } from '@vue/test-utils';
import Index from '@components/index.vue';
import Vue from 'vue';
import VeeValidate from 'vee-validate';

Vue.use(VeeValidate)

describe('Index.vue', () => {
    test('index matches snapshot', () => {
        const wrapper = mount(Index, {
            mocks: {
                $t: (key: string) => key,
            },
            stubs: {
                RouterLink: RouterLinkStub,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
