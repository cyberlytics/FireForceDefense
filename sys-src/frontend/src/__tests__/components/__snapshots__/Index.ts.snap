// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index.vue index matches snapshot 1`] = `
<div
  class="main-menu-bg"
>
  <div
    class="main-menu"
  >
    <div>
      <img
        alt="FireForceDefense"
        class="img-fluid"
        src="/assets/logo/logo.png"
        style="width: 250px;"
      />
    </div>
     
    <!---->
     
    <div
      class="input-group mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <label
          class="input-group-text"
          for="nickname"
        >
          Nickname
        </label>
      </div>
       
      <input
        class="form-control"
        id="nickname"
      />
    </div>
     
    <div
      class="mb-3"
    >
      <button
        class="btn btn-primary btn-block btn-lg"
        type="submit"
      >
        
                log in
            
      </button>
    </div>
     
    <hr
      class="mb-3"
    />
     
    <div>
      <a
        class="btn btn-primary btn-block btn-lg"
      >
        Credits
      </a>
    </div>
  </div>
</div>
`;
